packages/build/src/status/validations.ts

```diff
@@ -0,0 +1,24 @@
+import { DynamicMethods } from 'packages/js-client/lib/types.js'
+
+import { SecretScanResult } from '../plugins_core/secrets_scanning/utils.js'
+import { SystemLogger } from '../plugins_core/types.js'
+
+// Reports any validations completed on the deploy to the API
+export const reportValidations = async function ({
+  api,
+  secretScanResult,
+  deployId,
+  systemLog,
+}: {
+  api: DynamicMethods
+  secretScanResult: SecretScanResult
+  deployId: string
+  systemLog: SystemLogger
+}) {
+  try {
+    // @ts-expect-error Property 'updateDeployValidations' does not exist on type 'DynamicMethods'. This is a private/internal-only method and isn't generated in the type definitions.
+    await api.updateDeployValidations({ deploy_id: deployId, body: { secrets_scan: secretScanResult } })
+  } catch {
+    systemLog('Unable to report secrets scanning results to API')
```

Comment:
You may want to pass the caught error as well as perhaps a deploy id or other values that will help track or resolve the issue

Reply:
Yeah fair point, added 👍